airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx

```diff
@@ -112,6 +114,11 @@ export const AuthenticationProvider: React.FC = ({ children }) => {
         queryClient.removeQueries();
         loggedOut();
       },
+      async updateName(name: string): Promise<void> {
+        await userService.changeName(state.currentUser?.authUserId || "", state.currentUser?.userId || "", name);
+        updateUserName({ value: name });
+        return authService.updateProfile(name);
```

Comment:
Add safety checks here to ensure both firebase and our user service get updated, and if one of them fails, rollback and return. the userName state should be updated last when it's guaranteed that both the user service and firebse got the update.